File: java/src/main/java/FizzBuzz.java
import java.util.stream.IntStream;

/**
 * Created by vicboma on 25/11/15.
 */
public class FizzBuzz {

    public static final String EMPTY = "";
    public static final String FIZZ = "Fizz ";
    public static final String BUZZ = "Buzz ";
    public static final String CARRY = "\n";
    public static final int ZERO = 0;
    public static final int FIVE = 5;
    public static final int THREE = 3;
    public static final int INIT_RANGE = 1;
    public static final int END_RANGE_INCLUSIVE = 100;

    public FizzBuzz() {
    }

    public String execute(final StringBuilder sb){
        IntStream.rangeClosed(INIT_RANGE, END_RANGE_INCLUSIVE)
                 .boxed()
                 .forEach(x ->{
                     create(sb, x);
                 });
        return sb.toString();
    }

    private void create(StringBuilder sb, Integer x) {
        String str = EMPTY;
        str = fizz(x, str);
        str = buzz(x, str);
        str = carry(x, str);
        sb.append(str);
    }

    private String carry(Integer x, String str) {
        if (str == EMPTY)
            str += x;
        return str+CARRY;
    }

    private String buzz(Integer x, String str) {
        if(isMultiple(x,FIVE))
            str += BUZZ;
        return str;
    }

    private String fizz(Integer x, String str) {
        if(isMultiple(x,THREE))
            str += FIZZ;
        return str;
    }

    private Boolean isMultiple(Integer a, Integer b){
        return (a % b == ZERO);
    }
}

File: kotlin/src/main/kotlin/FizzBuzz.kt
/**
 * Created by vicboma on 25/11/15.
 */
class FizzBuzz(){

    companion object {
        val FIZZ = "Fizz"
        val BUZZ = "Buzz"
        val CARRY = "\n"
        val ZERO = 0
        val FIVE = 5
        val THREE = 3
        val INIT_RANGE = 1
        val END_RANGE_INCLUSIVE = 100
    }

    init {

    }

    public fun execute(sb : StringBuilder) : String {
        for(i in INIT_RANGE..END_RANGE_INCLUSIVE){
            sb.append(
                    when {
                        isMultiple(i,THREE) && isMultiple(i,FIVE) -> "$FIZZ$BUZZ"
                        isMultiple(i,THREE)  -> "$FIZZ"
                        isMultiple(i,FIVE)  -> "$BUZZ"
                        else -> "$i"
                    }+"$CARRY"
                )
            }
        return sb.toString()
    }


    private fun isMultiple(a : Int, b :Int) = (a % b) == ZERO
}